client/src/main/kotlin/io/spine/examples/chatspn/desktop/chat/ChatsPageModel.kt

```diff
@@ -61,36 +76,210 @@ public class ChatsPageModel {
     }
 
     /**
-     * Returns profile of the currently authorized user.
+     * Returns the state of selected chat.
+     */
+    public fun selectedChat(): StateFlow<ChatId> {
+        return selectedChatState
+    }
+
+    /**
+     * Selects provided chat.
      */
-    public fun authorizedUser(): UserProfile {
-        return authorizedUser!!
+    public fun selectChat(chat: ChatId) {
+        selectedChatState.value = chat
+        updateMessages(chat, client.readMessages(chat).toMessageDataList())
+        client.cancelMessagesSubscriptions()
+        client.subscribeOnMessages(
```

Comment:
This piece of code looks a bit difficult to read. Let's move and extract things, so that we could make `selectChat` method _much_ more readable.